.circleci/check-links/markdown-link-check

```diff
@@ -0,0 +1,165 @@
+            }
+            opts.baseUrl = url.format(parsed);
+        } catch (err) { /* ignore error */
+            }
+    } else {
+        fs.stat(filenameOrUrl, function(error , stats){
+            if (!error && stats.isDirectory()){
+                console.error(chalk.red('\nERROR: ' + filenameOrUrl + ' is a directory! Please provide a vaild filename as an argument.'));
+                process.exit(1);
+            }
+        });
+        opts.baseUrl = 'file://' + path.dirname(path.resolve(filenameOrUrl));
+        stream = fs.createReadStream(filenameOrUrl);
+    }
+
+}).parse(process.argv);
+
+opts.showProgressBar = (program.progress === true); // force true or undefined to be true or false.
+opts.quiet = (program.quiet === true);
+opts.verbose = (program.verbose === true);
+
+let markdown = ''; // collect the markdown data, then process it
+
+stream
+  .on('data', function (chunk) {
+    markdown += chunk.toString();
+  })
+  .on('error', function(error) {
+    if (error.code === 'ENOENT') {
+      console.error(chalk.red('\nERROR: File not found! Please provide a vaild filename as an argument.'));
+    } else {
+      console.error(chalk.red(error));
+    }
+    return process.exit(1);
+  })
+  .on('end', function () {
+    if (filenameForOutput) {
+        console.log(chalk.cyan('\nFILE: ' + filenameForOutput));
+    }
+
+    if (program.config) {
+        fs.access(program.config, (fs.constants || fs).R_OK, function (err) {
+            if (!err) {
+                let configStream = fs.createReadStream(program.config);
+                let configData = '';
+
+                configStream.on('data', function (chunk) {
+                    configData += chunk.toString();
+                }).on('end', function () {
+                    let config = JSON.parse(configData);
+
+                    opts.ignorePatterns = config.ignorePatterns;
+                    opts.replacementPatterns = config.replacementPatterns;
+                    opts.httpHeaders = config.httpHeaders;
+                    opts.ignoreDisable = config.ignoreDisable;
+
+                    runMarkdownLinkCheck(markdown, opts);
+                });
+            }
+            else {
+                console.error(chalk.red('\nERROR: Config file not accessible.'));
+                process.exit(1);
+            }
+        });
+    }
+    else {
+        runMarkdownLinkCheck(markdown, opts);
+    }
+});
+
+function runMarkdownLinkCheck(markdown, opts) {
+    markdownLinkCheck(markdown, opts, function (err, results) {
+        if (err) {
+            console.error(chalk.red('\nERROR: something went wrong!'));
+            console.error(err.stack);
+            process.exit(1);
+        }
+
+        if (results.length === 0 && !opts.quiet) {
+            console.log(chalk.yellow('No hyperlinks found!'));
+        }
+        results.forEach(function (result) {
+            // workaround to ignore 429 responses (too many requests)
```

Comment:
Is there any other way to change the behavior of the original markdown-link-check without copying and modifying its code? I want to avoid the additional maintenance burden if possible.

Reply:
This file primarily handles the command line arguments and prints the results from https://github.com/tcort/markdown-link-check/blob/v3.8.1/index.js, so none of the main processing logic is changed.  Also, maintenance should be minimal since I pinned the version of the original package.

Additionally, there is already a request in the upstream package to filter the results based on the response (https://github.com/tcort/markdown-link-check/issues/94), so this change may end up being a temporary workaround until that feature is implemented.